dotnet/src/SemanticKernel/SkillDefinition/NativeFunction.cs

```diff
@@ -160,17 +160,19 @@ public FunctionView Describe()
         CompleteRequestSettings? settings = null,
         CancellationToken cancellationToken = default)
     {
+#pragma warning disable CA1031 // Do not catch general exception types. The code will change in one of the next PRs to allow SK to throw exceptions instead of swallowing them. As a result, this suppression will be removed as well.
         try
         {
             return await this._function(null, settings, context, cancellationToken).ConfigureAwait(false);
         }
-        catch (Exception e) when (!e.IsCriticalException())
+        catch (Exception e)
```

Comment:
By removing the function `IsCriticalException` we are pratically considering any exceptions to be "critical" (should throw).

Two options: 
1. Remove all the catch block
2. Keep the catch just to log and `throw` after